aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -92,44 +92,92 @@ SparseTensor& mul_out_sparse_scalar(SparseTensor& r, const SparseTensor& t, Scal
 }
 
 // --------------------------------------------------------------------
-// log1p(SparseTensor)
+// Template unary op implementations
 // --------------------------------------------------------------------
 
-// In-place log1p on uncoalesced tensors is not supported since the operation is not a linear map.
-// Values of uncoalesced tensor corresponding to the same indices are summed
-// and log1p(summed_value) != log1p(v1) + log1p(v2)
+using FuncSig = Tensor& (Tensor::*)() const;
 
-SparseTensor& log1p_out_sparse(SparseTensor& r, const SparseTensor& t) {
+template<FuncSig F>
+SparseTensor& nonlinmap_unary_op_impl_out_sparse(SparseTensor& r, const SparseTensor& t, const std::string & err_msg)
+{
   TORCH_CHECK(r.is_sparse(), "Tensor should be sparse");
   TORCH_CHECK(t.is_sparse(), "Tensor should be sparse");
```

Comment:
Can the user trigger these errors with their input? If not, use `TORCH_INTERNAL_ASSERT` instead of `TORCH_CHECK`.